lib/WebService/GrowthBook/FeatureRepository.pm

```diff
@@ -0,0 +1,64 @@
+package WebService::GrowthBook::FeatureRepository;
+use strict;
+use warnings;
+no indirect;
+use Object::Pad;
+use HTTP::Tiny;
+use Log::Any qw($log);
+use Syntax::Keyword::Try;
+use JSON::MaybeUTF8 qw(decode_json_utf8);
+
+## VERSION
+
+class WebService::GrowthBook::FeatureRepository {
+    field $http :param :writer //= HTTP::Tiny->new();
+    method load_features($api_host, $client_key) {
+        # TODO add cache here
+        my $features = $self->_fetch_features($api_host, $client_key);
+        return $features;
+    }
+
+    method _fetch_features($api_host, $client_key){
+        my $decoded = $self->_fetch_and_decode($api_host, $client_key);
+        
+        # TODO decrypt here
+        if(exists $decoded->{features}){
+            return $decoded->{features};
+        }
+        else {
+            $log->warn("GrowthBook API response missing features");
+            return
+        }
+    }    
+    
+    method _fetch_and_decode($api_host, $client_key){
+        try {
+            my $r = $self->_get($self->_get_features_url($api_host), $client_key);
+            if($r->{status} >= 400){
```

Comment:
Just curious about this status check. Why do we check if the status code is greater than or equal to 400 ?

Reply:
because 4xx means that an error happened.